### PR TITLE
add script for building docker image

### PR DESCRIPTION
**Pull request by datanel**
The docker image is built if the working directory is in clean state, on master
- build `kisiodigital/stomptherabbit:latest` if there is no tag on HEAD
- otherwise `kisiodigital/stomptherabbit:${GIT_TAG}`

Bonus (later): inject the current commit when compiling

`DRY_RUN=true make build_docker_image` only builds the image
`make build_docker_image` build + push

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -3,29 +3,52 @@ PROJECTNAME := $(shell basename "$(PWD)")
...
+endif
+
+.PHONY: docker_logout
+docker_logout: ## Logout from dockerhub
+	$(info Logout from the Kisio Docker registry)
+	docker logout ${KISIO_DOCKER_REGISTRY}
+
+.PHONY: build_docker_image
+build_docker_image: docker_login ## Build docker image
+	$(info > Build docker image)
```
Comment (woshilapin) [inline Makefile:49]: Any reason why some `$(info ...)` have a `>` and some others don't?

**Review by woshilapin — DISMISSED** [review]
Very clean code, nice and easy to read and understand!

File: deploy/scripts/build.sh [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+# get the tag from git
+tag=$(git describe --tags --abbrev=0 2> /dev/null)
+has_tag=$(git tag --list --points-at HEAD)
+[ -z "${has_tag}" ] && tag=latest
+[ -z "${tag}" ] && Error "impossible to get a tag"
+
+
+# step 1: build docker images for git HEAD
+echo "Building $image_fullname"
+image_fullname="${docker_namespace}/${image}:${tag}"
```
Comment (pascal-vs) [inline deploy/scripts/build.sh:46]: `image_fullname` is not set for `echo`
```suggestion
image_fullname="${docker_namespace}/${image}:${tag}"
echo "Building ${image_fullname}"
```

File: deploy/scripts/build.sh [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+docker build --force-rm -t "${image_fullname}" "${script_path}"/../..
+
+# step 2: push the image to the registry
+if [ "${dry_run}" = 'false' ]; then
+    echo "pushing '${image_fullname}'"
+    docker push "${image_fullname[${image}]}" || Error "pushin ${image} failed"
```
Comment (pascal-vs) [inline deploy/scripts/build.sh:52]:
```suggestion
    docker push "${image_fullname}" || Error "pushing ${image} failed"
```

File: deploy/scripts/build.sh [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+# build only 'master' branch
+branch=$(git rev-parse --abbrev-ref HEAD 2> /dev/null)
+[ "${branch}" != 'master' ] && Error "building is allowed only on branch 'master'"
+
+# step 0: prepare docker image name with registry and tag
+# get the tag from git
+tag=$(git describe --tags --abbrev=0 2> /dev/null)
+has_tag=$(git tag --list --points-at HEAD)
+[ -z "${has_tag}" ] && tag=latest
```
Comment (pascal-vs) [inline deploy/scripts/build.sh:40]:
```suggestion
[ -z "${has_tag}" ] && tag='latest'
```

Comment (datanel) [reply deploy/scripts/build.sh:40]: why sometimes `""` and sometitmes `''` ?

File: Makefile (hunk 1 above)
Comment (datanel) [reply Makefile:49]: > is not important it's just to differentiate a comment from a command

**Comment by datanel** [comment]
everything fixd @woshilapin @pascal-vs

**Review by woshilapin — DISMISSED** [review]

File: Makefile [hunk 5]
```diff
@@ -3,29 +3,52 @@ PROJECTNAME := $(shell basename "$(PWD)")
...
-	$(info >  Running binary)
+run: ## Run the program
+	$(info > Running binary)
 	GOOS=linux GOARCH=$(GOARCH) $(GO) run -race ./cmd/$(PROJECTNAME)
 
+.PHONY: docker_login
+docker_login: ## Login to dockerhub
+ifdef INTERACTIVE
+	$(info > Login to dockerhub)
+	docker login
```
Comment (ArnaudOggy) [inline Makefile:37]: No `${KISIO_DOCKER_REGISTRY}` like for logout ?

Comment (datanel) [reply Makefile:37]: actually the registry is not necessary in docker logout. by it is dockerhub

**Review by ArnaudOggy — APPROVED** [review]

**Pull request merged**